### PR TITLE
fix(type): update v10 token styles

### LINKED ISSUE #11864: Update v10 type tokens in monorepo
**Checklist**

- [ ] Update root token values with the full change list for v10 in the [parent issue](https://github.com/carbon-design-system/carbon/issues/11813)

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #11864

#### Changelog
- updates type tokens for 
    - $expressive-heading-04
    - $expressive-heading-05
    - $productive-heading-05
    - $display-03
    - $display-04

#### Testing / Reviewing
- compare changes to v10 specs listed in original issue https://github.com/carbon-design-system/carbon/issues/11813

### CODE REVIEW

File: packages/type/src/styles.js [hunk 1]
```diff
@@ -131,7 +131,7 @@ export const productiveHeading03 = {
 
 export const productiveHeading04 = {
```
Comment (aagonzales) [inline packages/type/src/styles.js]: This change should have been for "productive-heading-**05**" not "04"

**Review by aagonzales — CHANGES_REQUESTED** [review]
Just one mix up.

✅ $expressive-heading-04
✅ $expressive-heading-05
❌ $productive-heading-05
✅ $display-03
✅ $display-04

**Review by aagonzales — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]
LGTM once tests pass with updated snaps

File: packages/type/scss/_scale.scss [hunk 2]
```diff
@@ -37,6 +37,30 @@ $carbon--type-scale: ();
...
+/// 76px = carbon--type-scale(15)
+/// 84px = carbon--type-scale(16)
+/// 92px = carbon--type-scale(17)
+///102px = carbon--type-scale(18)
+///112px = carbon--type-scale(19)
+///122px = carbon--type-scale(20)
+///132px = carbon--type-scale(21)
+///144px = carbon--type-scale(22)
+///156px = carbon--type-scale(23)
+
```
Comment (tay1orjones) [inline packages/type/scss/_scale.scss]:
```suggestion
```

**Pull request merged**